src/riak_kv_compile_tab.erl

```diff
@@ -71,35 +100,75 @@ is_compiling(BucketType) ->
         compiling_state() | notfound.
 get_state(BucketType) when is_binary(BucketType) ->
     case dets:lookup(?TABLE, BucketType) of
-        [{_,_,_,State}] ->
+        [{_,_,_,_,State}] ->
```

Comment:
The whole idea is to clean up these before we get into the code.  This is the "upgrade once" idea which then in the rest of the code we assume to be already upgraded.